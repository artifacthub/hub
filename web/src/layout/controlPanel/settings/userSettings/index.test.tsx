import { render } from '@testing-library/react';
import React from 'react';

import UserSettingsSection from './index';

jest.mock('../../../common/SectionPanel', () => (props: any) => <div>{props.defaultSection}</div>);

const defaultProps = {
  activeSection: 'user',
  onAuthError: jest.fn(),
};

describe('UserSettingsSection', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly', () => {
    const result = render(<UserSettingsSection {...defaultProps} />);

    expect(result.asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders profile section', () => {
      const { getByText } = render(<UserSettingsSection subsection="profile" {...defaultProps} />);

      expect(getByText('profile')).toBeInTheDocument();
    });

    it('renders subscriptions section', () => {
      const { getByText } = render(<UserSettingsSection subsection="subscriptions" {...defaultProps} />);

      expect(getByText('subscriptions')).toBeInTheDocument();
    });

    it('renders apiKeys section', () => {
      const { getByText } = render(<UserSettingsSection subsection="api-keys" {...defaultProps} />);

      expect(getByText('api-keys')).toBeInTheDocument();
    });

    it('renders webhooks section', () => {
      const { getByText } = render(<UserSettingsSection subsection="webhooks" {...defaultProps} />);

      expect(getByText('webhooks')).toBeInTheDocument();
    });

    it('renders profile as default section', () => {
      const { getByText } = render(<UserSettingsSection {...defaultProps} />);

      expect(getByText('profile')).toBeInTheDocument();
    });
  });
});
