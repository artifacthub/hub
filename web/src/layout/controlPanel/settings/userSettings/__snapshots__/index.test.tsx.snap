// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettingsSection renders correctly 1`] = `
<DocumentFragment>
  <div>
    profile
  </div>
</DocumentFragment>
`;
